src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -206,6 +260,82 @@ trait ScriptGenerators extends BitcoinSLogger {
     (p2SHScriptSignature, p2SHScriptPubKey, privateKeys)
   }
 
+  /**
+    * Generates a signed [[CLTVScriptSignature]] that spends from a [[CLTVScriptPubKey]] correctly
+    *
+    * @return the signed [[CLTVScriptSignature]], the [[CLTVScriptPubKey]] it spends, and the sequences of [[ECPrivateKey]]
+    *         used to sign the scriptSig
+    */
+  def signedCLTVScriptSignature(cltvLockTime : ScriptNumber, lockTime : UInt32, sequence : UInt32) : Gen[(ScriptSignature, CLTVScriptPubKey, Seq[ECPrivateKey])] = for {
+   // (cltv, privKey, num) <- cltvScriptPubKey(scriptNumber)
+    (scriptSig, scriptPubKey, privKeys) <- chooseSignedScriptSig
+    cltv = CLTVScriptPubKey(cltvLockTime, scriptPubKey)
+  } yield scriptPubKey match {
+    case _ : MultiSignatureScriptPubKey =>
+      val requiredSigs = scriptSig.signatures.size
+      val (cltvScriptSig, _, _) = cltvHelper(lockTime, sequence, cltv, privKeys, Some(requiredSigs))
+      (cltvScriptSig, cltv, privKeys)
+    case _ : P2PKHScriptPubKey | _ : P2PKScriptPubKey =>
+      val (cltvScriptSig, _, _) = cltvHelper(lockTime, sequence, cltv, privKeys, None)
+      (cltvScriptSig, cltv, privKeys)
+    case _ => throw new IllegalArgumentException("We only want to generate P2PK, P2PKH, and MultiSig ScriptSignatures.")
```

Comment:
Don't use a `_`, match on all of the scriptPubKey types. If in the future we could create new valid scriptPubKeys we need to take into consideration for `cltvScriptPubKey`